### ISSUE TITLE
enhance XML reader in fun.IO

### ISSUE THREAD

**Issue by UgiNyaa** [htl22-3ahif/fun#16]
The base that we build up isn't that bad, but we have to improve some things.
- There is a problem with setting properties. Now we are doing it like this:

``` xml
<Element type="ElementType" property0="value0" property1="value1" .../>
```

but to keep the attributes generic isn't the right way. I would like to do it like this:

``` xml
<Element type="ElementType">
  <Property name="property0" value="value0"/>
  <Property name="property1" value="value1"/>
  ...
</Element>
```
- The next problem deals with initializing the property. More specifically, how we initialize structs of classes, which may need more that one value. I would do it like that, but I'm not quite sure:

``` xml
<Element type="ElementType">
  <!--needs a vector3-->
  <Property name="property0" value="1/1/1"/>
  <!--needs a vector2-->
  <Property name="property1" value="1/1"/>
  ...
</Element>
```

so that we seperate each parameter with a "/" (or other not often used char)
- The third and last variation is to make another child node:

``` xml
<Element type="ElementType">
  <!--needs a vector3-->
  <Property name="property0">
    <Param type="float" value="1"/>
    <Param type="float" value="1"/>
    <Param type="float" value="1"/>
  </Property>
  <!--needs a vector2-->
  <Property name="property1">
    <Param type="float" value="1"/>
    <Param type="float" value="1"/>
  </Property>
  ...
</Element>
```

but I don't know if this would be too much. It could also cover the case, where the parameter of a struct/class is a struct/class:

``` xml
<Element type="ElementType">
  <!--needs a vector3-->
  <Property name="property0">
    <!--uses vector2 as parameter-->
    <Param type="Vector2">
      <Param type="float" value="1"/>
      <Param type="float" value="1"/>
    </Param>
    <Param type="float" value="1"/>
  </Property>
</Element>
```

This is it.
What do you think? Feel free to share your opinion. It would help me a lot.

**Comment by UgiNyaa** [comment]
I chose the 3rd one. You have more possibilitys with it. Also finished it, gonna do a commit.

**Comment by UgiNyaa** [comment]
Yo guys, I thought the enhancement of our xml reader is over, but it is not. After the writing problems occured, we gave up on xml and went to the standart binary serialize libarys by microsoft. I was not happy with it, but I also did not have any onther solution... unitl now.

I was experimenting with serialization since then and I got a new idea and tested it already:

``` xml
<Element Type="TransformElement">
      <Field Name="Position">
        <Field Name="X">0</Field>
        <Field Name="Y">0</Field>
        <Field Name="Z">5</Field>
      </Field>
      <Field Name="Scale">
        <Field Name="X">0,1</Field>
        <Field Name="Y">0,1</Field>
        <Field Name="Z">0,1</Field>
      </Field>
</Element>
```

And like that, we can also serialize our Elements without problems. Primitive, String, Class and Struct type is already supported, just need arrays, and we should be done for our current purposes.

So now I will explain how it exactly work. Earlier, we got the solution to write the parameters for the constructor of each type. That works fine, but when we go to the writer, he does not know which value belong to which parameter at the constructor. So this works only for deserializing our environment.

But now we use the philosophy that every object consists of fields (and ignoring the constructor), where all used informations are, so we only serialize and deserialize the fields (public and non public) that we get the object how it was before serializing. It is also readable, when you give the fields proper names. Now you just refer to the fields of your object and everything is ok.

I am also going to post the commit with the solutions after finishing some details. If you do not understand something or have questions, feel free to write them below.
